<launch>
<!-- Launch file for Velodyne16 VLP-16 LiDAR -->

	<arg name="rviz" default="true" />

	<node pkg="point_lio" type="pointlio_mapping" name="laserMapping" output="screen">
	<rosparam command="load" file="$(find point_lio)/config/unilidar.yaml" />
	<param name="prop_at_freq_of_imu" type="bool" value="1"/>	<!-- better be-->
	<param name="use_imu_as_input" type="bool" value="1"/>		<!-- would be. also as input of kalman filter-->
	<param name="check_satu" type="bool" value="1"/>
	<param name="init_map_size" type="int" value="10"/>			<!-- init map should have at least as such points-->
	<param name="point_filter_num" type="int" value="1"/>		<!-- msg points down sample: scan point selected every such adj--> 
	<param name="space_down_sample" type="bool" value="1" />	<!-- if need to down smapling-->
	<param name="filter_size_surf" type="double" value="0.5" />	<!-- down sampling for current sampling--> 
	<param name="filter_size_map" type="double" value="0.5" />	<!-- down sampling for ivox increment form downsampled pcl-->
	<param name="ivox_nearby_type" type="int" value="6" />		<!-- 0: Central; 6: 6-neibor; 18: 18-neibor; 26: 26-neibor-->
	<param name="runtime_pos_log_enable" type="bool" value="0" /> <!-- print consuming time for each module-->
	</node>
	<group if="$(arg rviz)">
	<node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find point_lio)/rviz_cfg/loam_livox.rviz" />
	</group>

	launch-prefix="gdb -ex run --args"

</launch>
